### ISSUE TITLE
Finetuned Regression

### ISSUE THREAD

**Issue by SkwisgaarSkwigelf** [yyly6/mRNABERT#5]
This may be a basic question, but could you provide example code on the proper way to use a finetuned regression model to predict a value? For example, I was able to successfully finetune on the mRFP dataset using the regression.py script, and got decent correlations. Now I would like to input a new RFP sequence and predict expression level.

I've tried to do this in the below code, but the value I'm getting is outside of the range of values in the training data (labels range from ~8-11 for the RFP dataset). In this example, I expect the finetuned model to predict a value of ~10.1 for the input sequence, however the output value is 3.06. I've also tried taking the top ~50 sequences from the training data, and the predicted value is always the same (~3.06) no matter the input, so I must be doing something wrong here. Any help would be appreciated, thank you!

```
import torch
from transformers import AutoTokenizer, AutoModelForSequenceClassification
from transformers.models.bert.configuration_bert import BertConfig

config = BertConfig.from_pretrained("/path/to/RFP_finetuned_regression_output/")
tokenizer = AutoTokenizer.from_pretrained("/path/to/RFP_finetuned_regression_output/")
model = AutoModelForSequenceClassification.from_pretrained("/path/to/RFP_finetuned_regression_output/")

#this is the first sequence from the train.csv file, ground truth label = 10.165
input_RFP_seq = 'ATG GCA TCA TCA GAA GAC GTC ATA AAA GAA TTT ATG CGA TTC AAA GTC CGC ATG GAA GGG TCA GTA AAC GGA CAC GAA TTC GAA ATC GAA GGA GAA GGA GAA GGC CGC CCA TAC GAA GGT ACC CAA ACA GCC AAA CTA AAA GTA ACT AAA GGC GGA CCA CTC CCC TTC GCC TGG GAC ATC CTC TCG CCC CAG TTC CAG TAC GGG TCT AAA GCC TAC GTC AAG CAC CCC GCC GAT ATT CCG GAT TAT CTG AAA CTA TCG TTT CCA GAG GGT TTT AAA TGG GAG CGG GTA ATG AAT TTT GAA GAC GGA GGG GTA GTA ACT GTT ACT CAA GAC TCA TCT CTG CAG GAT GGT GAA TTT ATC TAT AAG GTC AAA CTA CGA GGC ACC AAC TTC CCG TCG GAC GGA CCA GTC ATG CAA AAA AAA ACC ATG GGT TGG GAG GCT TCT ACC GAA CGC ATG TAC CCC GAG GAC GGC GCG CTC AAA GGC GAG ATC AAG ATG CGT CTC AAA CTC AAG GAC GGA GGT CAT TAT GAT GCG GAA GTT AAG ACT ACT TAC ATG GCA AAG AAA CCA GTC CAA CTA CCG GGA GCA TAT AAA ACT GAC ATA AAG CTT GAT ATT ACG TCG CAT AAT GAG GAT TAC ACT ATT GTG GAG CAG TAT GAG CGG GCC GAG GGC CGG CAT TCT ACC GGT GCG TGA'

inputs = tokenizer(input_RFP_seq, add_special_tokens=True, padding='longest', return_tensors="pt")

with torch.no_grad():
    model.eval()
    outputs = model(**inputs)
    logits = outputs.logits 
    predicted_expression = logits.squeeze().item()

print(predicted_expression)

```

**Comment by yyly6** [comment]
Getting a constant prediction like 3.06 instead of your expected 10.1 is a classic issue that occurs because your minimal script only loads the pre-trained base model with a randomly initialized regression head, completely missing the fine-tuned weights you trained. We have uploaded the predict.py script to meet your needs. To use it for your new RFP sequence, simply create a CSV file with headers like "sequence" and "label"（same as train.csv), place your sequence in the first column with a dummy label like 0 or just leave the second column blank in the second, update the --data_file and --output_file paths in your predict.sh file, and run the bash script to get your true predicted expression levels.  

```
python predict.py \
    --model_path /path/to/RFP_finetuned_regression_output/ \
    --data_file new_rfp_sequences.csv \
    --output_file my_new_predictions.csv \
    --max_length 1024 \
    --batch_size 8 \
    --device cuda 
```

[predict.py](https://github.com/user-attachments/files/25541303/predict.py)
